lib/pure/typetraits.nim

```diff
@@ -16,6 +16,18 @@ export system.`$` # for backward compatibility
 
 include "system/inclrtl"
 
+type
+  Typeid* = distinct string ## opaque, used by `getTypeid`
+proc getTypeid*(t: typedesc): Typeid {.magic: "TypeTrait", since: (1, 1).} =
```

Comment:
The names should be ``TypeId`` and ``getTypeId``. Also the documention could mention what a type ID is good for. (Comparisons? Store into sets to prevent recursions during type traversal?) Also it needs to mention that you cannot serialize these type IDs in any meaningful manner as after a recompile everything changes and IDs are **reused** in an unpredictable manner.

Reply:
> The names should be TypeId and getTypeId

done

> Also the documention could mention what a type ID is good for

done, I added detailed example use cases in ttypetraits.nim and referred to these in docs; these are prime motivations for this feature that were impossible/hard without this feature.

> Also it needs to mention that you cannot serialize these type IDs

done (a bit reworded)